environment.yml

```diff
@@ -5,20 +5,13 @@ channels:
   - conda-forge
 
 dependencies:
-  - python>=3.7
-  - hyperopt>=0.2.3
-  - matplotlib>=3.1.3
-  - numpy>=1.18.1
-  - pandas>=1.0.3
-  - pandas-flavor>=0.2.0
-  - pip>=20.0.2
-  - pytorch>=1.4.0
-  - rdkit>=2020.03.1.0
-  - scikit-learn>=0.22.2.post1
-  - scipy>=1.4.1
-  - tensorboardX>=2.0
-  - torchvision>=0.5.0
-  - tqdm>=4.45.0
-  - pip:
-    - typed-argument-parser>=1.6.1
-    - git+https://github.com/bp-kelley/descriptastorus
+  - python>=3.11
```

Comment:
Was the `pyproject.toml` file missing some depedencies before you copied it over to this file, or are we sure that all of those are no longer required in v2? e.g. I no longer see `hyperopt`, `matplotlib`, `pandas-flavor`, `tensorboardX`, `torchvision`, or `tqdm`. I would think we will at least need to retain `hyperopt` out of this list, although HPO has not yet been implemented for v2.

Reply:
Relatedly, is it okay that the minimum versions that were originally specified in this file have been removed for most packages that were retained? And that these minimum versions are also missing from the `pyproject.toml`?

Reply:
I checked and none of these removed dependencies are used explicitly in v2 as it now stands. I also am in favor of retaining hyperopt as that will be added in the not too distant future (though maybe it would make more sense to add it back as part of that PR). `pandas-flavor` and `torchvision` were not used explicitly in v1 best I can tell, so I don't know if they were/are needed implicitly.

Reply:
The toml and yml were not matched, and I don't know which was 'correct' (if either). As for the specific packages:
 - hyperopt: yes, will need to be added back if we re-implement hpo with hyperopt. Optuna may be a better option, since it plays well with Hydra (as does PL).
 - matplotlib was only used by the tsne script (in scripts), which has not been ported to v2. I also don't think it makes sense to package all of chemprop with matplotlib if only one script (that most people wont use) requires it
 - pandas-flavor is not used anywhere in our code, and if its a dependency of some of our other dependents its up to them to require it
See Nathan's comments about the other deps - this is all I had to add on.

For the package versions, since we moved to Python 3.11+ only versions of those dependencies which satisfy those requirements are eligible anyway.

Reply:
the `.yml` is way out of date- I think if anything it's the current `v1.6` env file, so I would ignore for `v2` purposes. If we were professionals, we would've all been using the same `.yml` for development purposes, but alas! I think we should probably raise this in another PR though and let it slide for now (i.e., not worry the `.yml` file)